javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -187,27 +88,31 @@ open class HandleManagerTestBase {
         storageKey = hatCollectionRefKey
     )
 
+    lateinit var schedulerProvider: JvmSchedulerProvider
     lateinit var readHandleManager: EntityHandleManager
     lateinit var writeHandleManager: EntityHandleManager
 
     open var testRunner = { block: suspend CoroutineScope.() -> Unit ->
-        runBlockingTest { this.block() }
+        runBlocking {
+            this.block()
+            schedulerProvider.cancelAll()
+        }
     }
 
     // Must call from subclasses.
     open fun setUp() {
         fakeTime = FakeTime()
         DriverAndKeyConfigurator.configure(null)
-        SchemaRegistry.register(Person)
-        SchemaRegistry.register(Hat)
-        DriverFactory.register(RamDiskDriverProvider())
+        RamDisk.clear()
     }
 
     // Must call from subclasses
-    open fun tearDown() {
-        RamDisk.clear()
-        DriverFactory.clearRegistrations()
-        SchemaRegistry.clearForTest()
+    open fun tearDown() = runBlocking {
+        schedulerProvider.cancelAll()
+        // TODO: this is less than ideal - we should investigate how to make the entire test process
+        //  cancellable/stoppable, even when we cross scopes into a BindingContext or over to other
+        //  RamDisk listeners.
+        delay(100) // Let things calm down.
```

Comment:
nit: If this is this likely cause flakyness? If so, maybe we should create a bug to track and fix it later?

Reply:
Will do!